spec/client_spec.rb

```diff
@@ -76,6 +76,36 @@ def redis.with
     context 'when lock is available' do
       after(:each) { lock_manager.unlock(@lock_info) if @lock_info }
 
+      context 'when redis connection error occurs' do
+        it 'locks if majority of redis instances are available' do
+          servers = [
+            RedisClient.new(url: "redis://#{redis1_host}:#{redis1_port}"),
+            RedisClient.new(url: "redis://#{redis2_host}:#{redis2_port}"),
+            unreachable_redis
+          ]
+
+          redlock = Redlock::Client.new(servers)
+
+          expect {
+            redlock.lock(resource_key, ttl)
+          }.to_not raise_error
+        end
+
+        it 'fails to acquire a lock if majority of redis instances are not available' do
```

Comment:
some respek, please:

```suggestion
        it 'fails to acquire a lock if majority of Redis instances are not available' do
```